meowth/__main__.py

```diff
@@ -3623,7 +3644,7 @@ def clean(txt):
                 teamcounts = teamcounts.lower().replace(word.lower(),"").replace(",","").strip()
     if teamcounts and teamcounts.split()[0].isdigit():
         total = int(teamcounts.split()[0])
-    elif ctx.author.id in trainer_dict:
+    elif (ctx.author.id in trainer_dict) and (trainer_dict[ctx.author.id]['status'] != None):
```

Comment:
the check for status is needed due to !cancel.
There are 3 ifs there in order of priority:
always use explicit value in the message.
if absent use data from existing trainer (basically, if trainer already updated their status, use that value)
else this is the first message from trainer, so do the math from the msg.

The issue is that !cancel resets count to 1 but doesn't clear the trainer from trainer dict.
Example: I send !c 1m 1i, which is accepted. Then i send !x. Then i resend !c 1m 1i. It says: Error: your team is 1 but the total is 2.
Since !x is iirc the only thing that sets the status to none, i'm saying "don't check trainer data if they sent !cancel"

Reply:
Fair enough. It should be in all three functions though.

Reply:
i'll look for the others.